File: mipush_hook/src/main/java/com/nihility/Dependencies.java
package com.nihility;

public class Dependencies {
    private static OuterDependencies outerDependencies;

    public static OuterDependencies instance() {
        return outerDependencies;
    }

    public static void set(OuterDependencies outerDependencies) {
        Dependencies.outerDependencies = outerDependencies;
    }

}

File: mipush_hook/src/main/java/com/nihility/Dependencies.kt
package com.nihility

class Dependencies {
    companion object {
        private var outerDependencies: OuterDependencies? = null

        @JvmStatic
        fun instance(): OuterDependencies? = outerDependencies

        @JvmStatic
        fun set(outerDependencies: OuterDependencies?) {
            Companion.outerDependencies = outerDependencies
        }
    }
}
